src/main/scala/com/fulcrumgenomics/alignment/Alignment.scala

```diff
@@ -278,4 +284,93 @@ case class Alignment(query: Array[Byte],
   }
+
+  /**
+    * Returns a subset of the Alignment representing the region defined by `start` and `end` on the query
+    * sequence. The returned alignment will contain the entire query and target sequences, but will have
+    * adjusted `queryStart` and `targetStart` positions and an updated `cigar`.  The score is set to 0.
+    *
+    * @param start the 1-based inclusive position of the first base on the query sequence to include
+    * @param end   the 1-based inclusive position of the last base on the query sequence to include
+    * @return a new [[Alignment]] with updated coordinates and cigar
+    */
+  def subByQuery(start: Int, end: Int): Alignment = {
+    require(start >= queryStart && start <= queryEnd, "start is outside of aligned region of target sequence")
+    require(end   >= queryStart && end   <= queryEnd, "end is outside of aligned region of target sequence")
+    sub(start, end, this.queryStart, _.operator.consumesReadBases())
+  }
+
+  /**
+    * Returns a subset of the Alignment representing the region defined by `start` and `end` on the target
+    * sequence. The returned alignment will contain the entire query and target sequences, but will have
+    * adjusted `queryStart` and `targetStart` positions and an updated `cigar`.  The score is set to 0.
+    *
+    * @param start the 1-based inclusive position of the first base on the target sequence to include
+    * @param end   the 1-based inclusive position of the last base on the target sequence to include
+    * @return a new [[Alignment]] with updated coordinates and cigar
+    */
+  def subByTarget(start: Int, end: Int): Alignment = {
+    require(start >= targetStart && start <= targetEnd, "start is outside of aligned region of target sequence")
+    require(end   >= targetStart && end   <= targetEnd, "end is outside of aligned region of target sequence")
+    sub(start, end, this.targetStart, _.operator.consumesReferenceBases())
+  }
+
+  /**
+    * Private helper method that helps generate an Alignment that is a subset of the current alignment.
+    *
+    * @param start the start (on either the query or target sequence) of the desired window
+    * @param end the end (on either the query or target sequence) of the desired window
+    * @param initialStart the start position on the relevant sequence (either target OR query)
+    * @param consumes a function that returns true if the cigar element passed as a parameter consumes
+    *                 bases on the sequence on which `start` and `end` are expresssed
+    * @return a new [[Alignment]] with adjusted start, end and cigar, and with score=0
+    */
+  private def sub(start: Int, end: Int, initialStart: Int, consumes: CigarElem => Boolean): Alignment = {
+    val elems = new ArrayBuffer[CigarElem](cigar.size)
+    var (qStart, tStart, currStart) = (queryStart,targetStart,initialStart)
+
+    cigar.foreach { elem =>
+      val currEnd = if (consumes(elem)) currStart + elem.length - 1 else currStart
```

Comment:
Just for our future sake, could you add a comment similar to: `// the end of the current element`?

Reply:
Also, do you want to save `consumes(elem)` to a `val` since you use it multiple times below?

Reply:
Done & Done